### PR TITLE
Ref/objfcts

### PR DESCRIPTION
**Pull request by lheagy**
- ensure that all children of a combo regularization are looking at the exact same regmesh, mappings
- do not store both the mesh and the regmesh on the regularization. Only the regmesh is needed. 
- add tests to ensure children of a combo regularization are looking at the same regmesh and mapping objects in memory

### CODE REVIEW

File: tests/base/test_regularization.py [hunk 1]
```diff
@@ -302,5 +303,35 @@ def test_update_of_sparse_norms(self):
...
 
+    def test_linked_properties(self):
+        mesh = Mesh.TensorMesh([8, 7, 6])
+        reg = Regularization.Tikhonov(mesh)
+
+        [self.assertTrue(reg.regmesh is fct.regmesh) for fct in reg.objfcts]
+        [self.assertTrue(reg.mapping is fct.mapping) for fct in reg.objfcts]
+
+        D = reg.regmesh.cellDiffx
+        reg.regmesh._cellDiffx = 4*D
```
Comment (lheagy) [inline tests/base/test_regularization.py]: @fourndo: this tests a use-case similar to the one that you showed with updating the cell Diffs

File: SimPEG/Regularization.py [hunk 2]
```diff
@@ -470,58 +459,6 @@ def _nC_residual(self):
         else:
             return self.nP
 
-    @property
-    def mesh(self):
```
Comment (rowanc1) [inline SimPEG/Regularization.py]: This is still accessible from `regmesh.mesh`...

Comment (lheagy) [inline SimPEG/Regularization.py]: yep. But I don't think there is any need to access it from the regularization itself, that is just a potential point on confusion

**Comment by fourndo** [comment]
Good job @lheagy . I merged in my local RotatedObjFun branch in it worked magically.
I vote yes.

**Pull request merged**